src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -116,27 +121,32 @@
 
     private final Object classVariableDefinitionLock = new Object();
 
+    private boolean preInitializing;
     private boolean initialized;
     private volatile boolean finalizing;
 
+    private static boolean preInitializeContexts = System.getProperty("polyglot.engine.PreinitializeContexts") != null;
```

Comment:
We're reading this for a second time. And should we be reading it as a boolean, what if it's set to `false`?

Reply:
It's a list of languages (e.g. `-Dpolyglot.engine.PreinitializeContexts=ruby`).
It doesn't matter to re-read since this is set at VM startup time.
Are you worried this takes time during startup?

Reply:
It's fine as long as we're not doing it twice.

Reply:
Done.